dotnet/src/Connectors/Connectors.AssemblyAI/AudioToText/AssemblyAIAudioToTextService.cs

```diff
@@ -0,0 +1,292 @@
+    /// <returns></returns>
+    public async Task<TextContent> GetTextContentAsync(
+        FileInfo file,
+        PromptExecutionSettings? executionSettings = null,
+        Kernel? kernel = null,
+        CancellationToken cancellationToken = default)
+    {
+        string uploadUrl;
+        using (var fileStream = file.OpenRead())
+        {
+            uploadUrl = await this.UploadFileAsync(fileStream, cancellationToken).ConfigureAwait(false);
+        }
+
+        var transcriptId = await this.CreateTranscriptAsync(uploadUrl, executionSettings, cancellationToken)
+            .ConfigureAwait(false);
+        var transcript = await this.WaitForTranscriptToProcessAsync(transcriptId, cancellationToken)
+            .ConfigureAwait(false);
+
+        return new TextContent(
+            text: transcript.RootElement.GetProperty("text").GetString(),
+            modelId: null,
+            // TODO: change to typed object when AAI SDK is shipped
+            innerContent: transcript,
+            encoding: Encoding.UTF8,
+            metadata: null
+        );
+    }
+
+    private async Task<string> UploadFileAsync(Stream audioStream, CancellationToken ct)
+    {
+        const string URL = "https://api.assemblyai.com/v2/upload";
+        using var content = new StreamContent(audioStream);
+        content.Headers.ContentType = new MediaTypeHeaderValue("application/octet-stream");
+        using var request = new HttpRequestMessage(HttpMethod.Post, URL);
+        request.Headers.Authorization = new AuthenticationHeaderValue(this._apiKey);
+        request.Content = content;
+        using var response = await this._httpClient.SendAsync(request, ct).ConfigureAwait(false);
+        await ThrowIfNotSuccessStatusCodeAsync("Failed to upload file.", response, ct)
+            .ConfigureAwait(false);
+        var jsonStream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+        var json = await JsonDocument.ParseAsync(jsonStream, cancellationToken: ct).ConfigureAwait(false);
+        return json.RootElement.GetProperty("upload_url").GetString()
+               ?? throw new AssemblyAIApiException("Property 'upload_url' expected but not found.");
+    }
+
+    private async Task<string> CreateTranscriptAsync(
+        string audioUrl,
+        PromptExecutionSettings? executionSettings = null,
+        CancellationToken ct = default
+    )
+    {
+        const string URL = "https://api.assemblyai.com/v2/transcript";
+        var jsonRequest = new JsonObject();
+        jsonRequest["audio_url"] = audioUrl;
+        if (executionSettings?.ExtensionData is not null)
+        {
+            foreach (var attribute in executionSettings.ExtensionData)
+            {
+                jsonRequest[attribute.Key] = JsonValue.Create(attribute.Value);
+            }
+        }
+
+        using var content = new StringContent(jsonRequest.ToJsonString(), Encoding.UTF8, "application/json");
```

Comment:
Look for class HttpRequest, there are ready method to create http request with payload.
Example from my code
```cs
var request = HttpRequest.CreatePostRequest(endpoint, requestData);
        request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", this._apiKey);
        request.Headers.Add("User-Agent", HttpHeaderConstant.Values.UserAgent);
        request.Headers.Add(HttpHeaderConstant.Names.SemanticKernelVersion,
            HttpHeaderConstant.Values.GetAssemblyVersion(typeof(OpenAIHttpRequestFactory)));
```